ecoscope/base/base.py

```diff
@@ -516,6 +518,165 @@ def turn_angle(traj):
 
         return angles.rename("turn_angle").reindex(self.index)
 
+    def upsample(self, upsample_time):
```

Comment:
Consider renaming `upsample_time` to `freq` and changing type from `int` to `pd.Timedelta` to match pandas.

Reply:
I think it's easier for a user that doesn't have much programming knowledge to use seconds (int) than a Timedelta object.